packages/vaadin-grid/test/row-details.test.js

```diff
@@ -297,4 +291,69 @@ describe('row details', () => {
       expect(row.offsetHeight).to.be.above(70);
     });
   });
+
+  describe('details opened attribute', () => {
+    let dataset = [];
+    const dataProvider = (params, callback) => callback(dataset);
+
+    const countRowsMarkedAsDetailsOpened = (grid) => {
+      return grid.$.items.querySelectorAll('tr[details-opened]').length;
+    };
+
+    beforeEach(async () => {
+      dataset = buildDataSet(10);
+      grid = fixtureSync(`
+        <vaadin-grid style="width: 50px; height: 400px" size="100">
+          <template class="row-details"><span>[[index]]</span>-details</template>
+          <vaadin-grid-column>
+            <template>[[index]]</template>
+          </vaadin-grid-column>
+        </vaadin-grid>
+      `);
+      grid.dataProvider = dataProvider;
+      flushGrid(grid);
+      bodyRows = getRows(grid.$.items);
+      await nextFrame();
+    });
+
+    it('should update when opening/closing imperatively', () => {
+      openRowDetails(1);
+      expect(bodyRows[1].hasAttribute('details-opened')).to.be.true;
+      expect(countRowsMarkedAsDetailsOpened(grid)).to.equal(1);
+      closeRowDetails(1);
+      expect(bodyRows[1].hasAttribute('details-opened')).to.be.false;
+      expect(countRowsMarkedAsDetailsOpened(grid)).to.equal(0);
+    });
+
+    it('should be removed when item is removed', () => {
+      openRowDetails(0);
+      dataset.shift(); // remove opened item
+      grid.clearCache();
+
+      expect(bodyRows[0].hasAttribute('details-opened')).to.be.false;
+      expect(countRowsMarkedAsDetailsOpened(grid)).to.equal(0);
```

Comment:
This could be redundant, but when trying this in the flow IT it seemed that the rows could be reused / shuffled in an unordered manner. As such I thought it would be a good idea to verify the outcome using the number of rows rather than expecting a specific row to be changed.